### PR TITLE
Add bug fixes to slalpha5 branch

### PR DESCRIPTION
**Pull request by aneeshafedo**
## Purpose

this PR addresses following issues
> Update & delete operations does not return the original response in success 
> Add Enums to identified collections of constants
> Revamp display annotations

Resolves  https://github.com/wso2-enterprise/choreo/issues/4518, 
Resolves https://github.com/wso2-enterprise/choreo/issues/4522,
Resolves https://github.com/wso2-enterprise/choreo/issues/4520

Also includes few spelling mistakes in documentation and client_endpoint.bal 

## Automation tests
 - Unit tests 
   > Updated
 - Integration tests
   > Updated

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -232,7 +232,7 @@ json account = {
        BillingCity: "Jaffna",
        Phone: "+94110000000"
    };
-boolean|sfdc:Error isSuccess = baseClient->updateRecord(ACCOUNT, testRecordId, account);
+boolean|sfdc:Error isSuccess = baseClient->updateRecord("Account", testRecordId, account);
```
Comment (Chuhaa) [inline README.md:235]: Check the return type.

File: README.md [hunk 2]
```diff
@@ -241,7 +241,7 @@ SObject Name and the SObject record id as parameters and the function will retur
 
 ```ballerina
 string testRecordId = "001xa000003DIlo";
-boolean|sfdc:Error isDeleted = baseClient->deleteRecord(ACCOUNT, testRecordId);
+boolean|sfdc:Error isDeleted = baseClient->deleteRecord("Account", testRecordId);
```
Comment (Chuhaa) [inline README.md:244]: Check the return type.

File: sfdc/client_endpoint.bal [hunk 3]
```diff
@@ -557,13 +554,13 @@ public client class Client {
     # + operation - type of operation like insert, delete, etc.
     # + sobj - kind of sobject 
     # + contentType - content type of the job 
-    # + extIdFieldName - field name of the external ID incase of an Upsert operation
+    # + extIdFieldName - field name of the external Id incase of an Upsert operation
     # + return - returns job object or error
     @display {label: "Create job"}
-    isolated remote function creatJob(@display {label: "Operation"} OPERATION operation, 
+    isolated remote function createJob(@display {label: "Operation"} Operation operation,
```
Comment (aneeshafedo) [inline sfdc/client_endpoint.bal:560]: Method name changed due to spelling mistake.

**Review by abeykoon — APPROVED** [review]

**Review by abeykoon — APPROVED** [review]

File: sfdc/client_endpoint.bal (hunk 3 above)
Comment (abeykoon) [reply sfdc/client_endpoint.bal:560]: Ack

**Pull request merged**